lib/apiservers/engine/backends/container.go

```diff
@@ -367,48 +438,124 @@ func (c *Container) ContainerStop(name string, seconds int) error {
 	return nil
 }
 
+// ContainerUnpause unpauses a container
 func (c *Container) ContainerUnpause(name string) error {
 	return fmt.Errorf("%s does not implement container.ContainerUnpause", c.ProductName)
 }
 
+// ContainerUpdate updates configuration of the container
 func (c *Container) ContainerUpdate(name string, hostConfig *container.HostConfig) ([]string, error) {
 	return make([]string, 0, 0), fmt.Errorf("%s does not implement container.ContainerUpdate", c.ProductName)
 }
 
+// ContainerWait stops processing until the given container is
+// stopped. If the container is not found, an error is returned. On a
+// successful stop, the exit code of the container is returned. On a
+// timeout, an error is returned. If you want to wait forever, supply
+// a negative duration for the timeout.
 func (c *Container) ContainerWait(name string, timeout time.Duration) (int, error) {
 	return 0, fmt.Errorf("%s does not implement container.ContainerWait", c.ProductName)
 }
 
 // docker's container.monitorBackend
 
+// ContainerChanges returns a list of container fs changes
 func (c *Container) ContainerChanges(name string) ([]archive.Change, error) {
 	return make([]archive.Change, 0, 0), fmt.Errorf("%s does not implement container.ContainerChanges", c.ProductName)
 }
 
+// ContainerInspect returns low-level information about a
+// container. Returns an error if the container cannot be found, or if
+// there is an error getting the data.
 func (c *Container) ContainerInspect(name string, size bool, version version.Version) (interface{}, error) {
-	return nil, fmt.Errorf("%s does not implement container.ContainerInspect", c.ProductName)
+	//Ignore version.  We're supporting post-1.20 version.
+
+	defer trace.End(trace.Begin("ContainerInspect"))
+
+	// Look up the container info in the metadata cache
+	vc := viccontainer.GetCache().GetContainerByName(name)
+	if vc == nil {
+		return nil, derr.NewRequestNotFoundError(fmt.Errorf("No such container: %s", name))
+	}
+
+	// HACK: Just set a bunch of dummy info needed to get docker attach working
```

Comment:
I think we need a FIXME here to indicate required future work, HACK sounds like we are done with it :)